lib/metasploit/framework/data_service/proxy/host_data_proxy.rb

```diff
@@ -88,4 +100,25 @@ def valid(opts)
     return true
   end
 
+  def use_id_or_get_id_from_host_address(opts, method)
```

Comment:
I really don't like the name for this method at all, it's not very clear what it's purpose is and it's being used to make changes like adding and deleting when it's implied it's only used to retrieve
I think it would be much clearer if this was refactored into a method that returns the id from the address then the caller can choose what to do with that information

Reply:
Yeah I'm with you, the method's doing too much and all I can say about the name is that it made sense at the time 🙃